### PR TITLE
openssf-compiler-options: add -Wno-hardened to avoid causing problems…

### PR DESCRIPTION
**Pull request by justinvreeland**
… with redefined FORTIFY_SOURCE

Remove lines adding this manually from ruby.

### CODE REVIEW

**Comment by justinvreeland** [comment]
@xnox Thoughts? This is a persistent issue I find and some things like python packages (in particular py3-ml-metadata) seem to go out of their way to ignore options passed other ways.

**Comment by xnox** [comment]
> @xnox Thoughts? This is a persistent issue I find and some things like python packages (in particular py3-ml-metadata) seem to go out of their way to ignore options passed other ways.

Yes please, as that's what Wno-error=hardened was supposed to do.

Unless I got the syntax wrong.

Plus many of them are false alarms too. Not adding blah because it was already specified to the same safe value because us or upstream were already doing best practice.

**Comment by justinvreeland** [comment]
> Yes please, as that's what Wno-error=hardened was supposed to do.
Yeah judging from where I see this problem that turns off the error but it still keeps it as a warning then if they have `-Werr` the warning will cause an error : (. There's also a fair amount of failure from additional configure output which need to be entirely suppressed to work correctly.

File: gcc-13.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ environment:
       - make
       - mpc-dev
       - mpfr-dev
+      - openssf-compiler-options
       - patch
       - texinfo
```
Comment (xnox) [inline gcc-13.yaml:33]: Is this hunk intentional? Why no epoch bump?

**Review by xnox — DISMISSED** [review]

File: gcc-13.yaml (hunk 1 above)
Comment (justinvreeland) [reply gcc-13.yaml:33]: This is intentional.  I initially had more extensive gcc-13 changes but merged it into here to build successfully. I'll bump it.

Comment (justinvreeland) [reply gcc-13.yaml:33]: Actually I'll remove it and fix it in the gcc pr I already have.

**Review by xnox — APPROVED** [review]

**Pull request merged**